ostree_upload_server/importers/tar.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import tempfile
+import tarfile
+
+import urllib
+import urlparse
+
+from gi.repository import GLib
+
+from os import path
+from shutil import rmtree
+
+from .base import BaseImporter
+from .util import find_repo, open_repository
+
+class TarImporter(BaseImporter):
+    MIME_TYPE = 'application/x-tar'
+
+    def _import_commit(self, target_repo, src_path_obj, refs=None):
+        if not self._source_repo_path:
+            raise RuntimeError("Cannot invoke _import_commit without calling import_to_repo first")
+
+        if not refs:
+            raise RuntimeError("Cannot invoke _import_commit without valid refs")
+
+        logging.debug("Importing %s%s into %s...", self._source_repo_path,
+                      refs,
+                      target_repo.get_path().get_uri())
+
+        options = GLib.Variant('a{sv}', {
+            'refs': GLib.Variant('as', refs),
```

Comment:
Don't pull by ref here since it will overwrite the ref in the target repo when pulling. Then there would be a short time where the ref will point to the temporary commit. Instead, specify the refs as the commit checksums like `'refs': GLib.Variant('as', [commit])`.

Reply:
Oh interesting - I didn't realize that a non-committed transaction changes the repo that way. Will change.

Reply:
Oh, right this is in the same transaction. Still, I think it's safer to pull by checksum instead of relying on the ref to be overwritten later.